src/rules/spaceWithinParensRule.ts

```diff
@@ -0,0 +1,167 @@
+        if (!isNaN(parsedSize)) {
+            size = parsedSize;
+        }
+    }
+    return {
+        size,
+    };
+}
+
+class SpaceWhitinParensWalker extends Lint.AbstractWalker<Options> {
+    private openingParenRegex = /^([^\(]*\()( *)[^\n\r]*$/m;
+    private closingParenRegex = /([^\s])(\s*)\)$/;
+
+    public walk(sourceFile: ts.SourceFile) {
+        const cb = (node: ts.Node): void => {
+            this.checkNode(node);
+            ts.forEachChild(node, cb);
+        };
+        return ts.forEachChild(sourceFile, cb);
+    }
+
+    private getOpenParenToken(node: ts.Node): ts.Node | undefined {
+        return Lint.childOfKind(node, ts.SyntaxKind.OpenParenToken);
+    }
+
+    private getCloseParenToken(node: ts.Node): ts.Node | undefined {
+        return Lint.childOfKind(node, ts.SyntaxKind.CloseParenToken);
+    }
+
+    private checkNode(node: ts.Node): void {
+        let child: ts.Node | undefined;
+        if ((child = this.getOpenParenToken(node)) !== undefined) {
+            this.checkOpenParenToken(child);
+        }
+        if ((child = this.getCloseParenToken(node)) !== undefined) {
+            this.checkCloseParenToken(child);
+        }
+    }
+
+    private checkOpenParenToken(node: ts.Node) {
+        const code = this.sourceFile.text.substr(node.getFullStart());
+        const matchResult = this.openingParenRegex.exec(code);
+        if (matchResult) {
+            const whitespace = matchResult[2];
+            const whitespaceLength = whitespace.length;
+            if (whitespaceLength !== this.options.size) {
+                const skipChars = matchResult[1].length;
+                let length = whitespaceLength;
+                let position = node.getFullStart() + skipChars;
+                if (length > this.options.size) {
+                    length -= this.options.size;
+                    position += this.options.size;
+                }
+                this.checkAndAddFailureAt(position, length, whitespaceLength);
+            }
+        }
+    }
+
+    private checkCloseParenToken(node: ts.Node) {
+        const code = this.sourceFile.text.substring(node.getFullStart() - 20, node.getEnd());
+        const matchResult = this.closingParenRegex.exec(code);
+        if (matchResult) {
+            const whitespace = matchResult[2];
+            const whitespaceLength = whitespace.length;
+            if (whitespaceLength !== this.options.size) {
+                let length = whitespaceLength;
+                const position = node.getEnd() - 1 - whitespaceLength;
+                if (length > this.options.size) {
+                    length -= this.options.size;
+                }
+                this.checkAndAddFailureAt(position, length, whitespaceLength);
+            }
+        }
+    }
+
+    private checkAndAddFailureAt(position: number, length: number, spaceCount: number = 0) {
+        if (this.failures.some(f => f.getStartPosition().getPosition() === position)) {
```

Comment:
in my sample implementation there was a check if `ch` is a line break to avoid failures.

When checking the CloseBrace, you could add a similar check if `ch === 40` to not fail if you reached the OpenBraceToken.